gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -770,21 +762,25 @@ gst_tensor_src_iio_get_all_channel_info (GstTensorSrcIIO * self,
       /** find and set the type information */
       filename = g_strdup_printf ("%s%s", channel_prop.base_file, TYPE_SUFFIX);
       if (!g_file_test (filename, G_FILE_TEST_IS_REGULAR)) {
-        channel_prop.generic_name =
-            gst_tensor_src_iio_get_generic_name (channel_prop.name);
-        silent_debug ("Generic name = %s", channel_prop.generic_name);
+        /** if specific type info unavailable, use generic type info */
         g_free (filename);
-        filename =
+        gchar *generic_type_filename =
```

Comment:
To be C89 compliant, you'll need to declare variable on the top of a code block.

Reply:
Done.